app/models/concerns/foreman_wreckingball/host_extensions.rb

```diff
@@ -21,5 +21,29 @@ module HostExtensions
     def action_input_key
       'host'
     end
+
+    def deduced_vsphere_guest
+      return unless architecture && operatingsystem
+
+      selectors = {
+        architecture: architecture.name,
+        osfamily: operatingsystem.family,
+        name: operatingsystem.name,
+        major: operatingsystem.major.to_i
+      }
+
+      deduced_os = VsphereOsIdentifiers.find_by(selectors)
+      deduced_os ||= VsphereOsIdentifiers.find_by(selectors.except(:major))
+      deduced_os ||= VsphereOsIdentifiers.find_by(selectors.except(:release))
+      deduced_os ||= VsphereOsIdentifiers.find_by(selectors.except(:major, :release))
+      deduced_os ||= VsphereOsIdentifiers.find_by(selectors.except(:major, :name))
+      deduced_os ||= VsphereOsIdentifiers.find_by(selectors.except(:major, :name, :release))
```

Comment:
Selector release does not exist anymore if I understand correctly. If it is ok, that it was removed lines 37, 38,40 are to be removed. If not it should be readded after line 32.